core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/AlertDialogShower.kt

```diff
@@ -51,6 +69,33 @@ class AlertDialogShower @Inject constructor(private val activity: Activity) :
               ?.invoke()
           }
         }
+        url?.let {
+          val textView = TextView(activity.baseContext)
+          textView.setPadding(5, 5, 5, 5)
+          textView.gravity = Gravity.CENTER
+          textView.setLinkTextColor(Color.BLUE)
+          textView.setOnLongClickListener {
+            val clipboard =
+              ContextCompat.getSystemService(activity.baseContext, ClipboardManager::class.java)
+            val clip = ClipData.newPlainText("External Url", "$url")
+            clipboard?.setPrimaryClip(clip)
+            Toast.makeText(
+              activity.baseContext,
+              R.string.external_link_copied_message,
+              Toast.LENGTH_SHORT
+            ).show()
+            true
+          }
+          textView.text = Html.fromHtml("</br><a href=$url> <b>$url</b>")
+
+          setView(
+            textView,
+            externalLinkLeftMargin,
+            externalLinkTopMargin,
+            externalLinkRightMargin,
+            externalLinkBottomMargin
+          )
+        }
```

Comment:
This method is deprecated. We can use the alternative one. Also, we can get rid of this restricted API warning.

Reply:
@gouri-panda , i have removed this deprecated method and restricted API warning.